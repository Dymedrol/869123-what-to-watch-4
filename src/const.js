const Genres = {
  ALL: `All genres`,
  COMEDIES: `Comedies`,
  CRIME: `Crime`,
  DOCUMENTARY: `Documentary`,
  DRAMA: `Drama`,
  HORROR: `Horror`,
  KIDS_FAMILY: `Kids & Family`,
  ROMANCE: `Romance`,
  SCI_FI: `Sci-Fi`,
  THRILLERS: `Thrillers`,
};

const Tabs = [
  `Overview`,
  `Details`,
  `Reviews`
];

const MovieListStep = {
  MAIN: 8,
  MOVIEPAGE: 4,
};

const videoPlayerModes = {
  SMALL: `small`,
  FULLSCREEN: `fullscreen`,
};

export {Genres, Tabs, MovieListStep, videoPlayerModes};
