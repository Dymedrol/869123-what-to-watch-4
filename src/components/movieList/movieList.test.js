import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";

import MovieList from "./movieList.jsx";
import NameSpace from "../../reducer/nameSpace.js";
import {Genres} from "../../const.js";


const mockStore = configureStore([]);

const movies = [{
  name: `Title`,
  previewImage: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
}];

it(`Проверка снепшота компонента MovieList`, () => {
  const store = mockStore({
    [NameSpace.APP]: {
      genre: Genres.ALL,
    },
    [NameSpace.DATA]: movies,
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <MovieList
            movies={movies}
            setActiveItem={() => {}}
            removeActiveItem={() => {}}
          />
        </Provider>, {
          createNodeMock: () => {
            return {};
          }})
    .toJSON();

  expect(tree).toMatchSnapshot();
});
