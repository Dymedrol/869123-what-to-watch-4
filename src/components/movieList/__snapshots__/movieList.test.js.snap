// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверка снепшота компонента MovieList 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        className="player__video"
        poster="fantastic-beasts-the-crimes-of-grindelwald.jpg"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Title
      </a>
    </h3>
  </article>
</div>
`;
